R/model.R

```diff
@@ -1821,6 +1822,124 @@ variational <- function(data = NULL,
+#'   implements two variational algorithms. The `algorithm="meanfield"` option
+#'   uses a fully factorized Gaussian for the approximation. The
+#'   `algorithm="fullrank"` option uses a Gaussian with a full-rank covariance
+#'   matrix for the approximation.
+#'
+#'   -- [*CmdStan Interface User's Guide*](https://github.com/stan-dev/cmdstan/releases/latest)
+#'
+#' @template model-common-args
+#' @param num_threads (positive integer) If the model was
+#'   [compiled][model-method-compile] with threading support, the number of
+#'   threads to use in parallelized sections (e.g., for multi-path pathfinder
+#'   as well as `reduce_sum`)
+#' @param init_alpha (positive real) The initial step size parameter.
+#' @param tol_obj (positive real) Convergence tolerance on changes in objective function value.
+#' @param tol_rel_obj (positive real) Convergence tolerance on relative changes in objective function value.
+#' @param tol_grad (positive real) Convergence tolerance on the norm of the gradient.
+#' @param tol_rel_grad (positive real) Convergence tolerance on the relative norm of the gradient.
+#' @param tol_param (positive real) Convergence tolerance on changes in parameter value.
+#' @param history_size (positive integer) The size of the history used when
+#'   approximating the Hessian.
+#' @param num_draws (positive integer) Number of draws to return after performing pareto smooted importance sampling (PSIS)
+#' @param num_paths (positive integer) Number of single pathfinders to run
+#' @param max_lbfgs_iters (positive integer) The maximum number of iterations for LBFGS
+#' @param num_elbo_draws (positive integer) Number of draws to make when calculating the ELBO of
+#'   the approximation at each iteration of LBFGS
+#' @param save_single_paths (logical) Whether to save the results of single pathfinder runs in multi-pathfinder
+#' @return A [`CmdStanPathfinder`] object.
+#'
+#' @template seealso-docs
+#' @inherit cmdstan_model examples
+#'
+pathfinder <- function(data = NULL,
+                        seed = NULL,
+                        refresh = NULL,
+                        init = NULL,
+                        save_latent_dynamics = FALSE,
+                        output_dir = NULL,
+                        output_basename = NULL,
+                        sig_figs = NULL,
+                        opencl_ids = NULL,
+                        num_threads = NULL,
+                       init_alpha = NULL,
+                       tol_obj = NULL,
+                       tol_rel_obj = NULL,
+                       tol_grad = NULL,
+                       tol_rel_grad = NULL,
+                       tol_param = NULL,
+                       history_size = NULL,
+                       num_draws = NULL,
+                       num_paths = NULL,
+                       max_lbfgs_iters = NULL,
+                       num_elbo_draws = NULL,
+                       save_single_paths = NULL) {
+  procs <- CmdStanProcs$new(
+    num_procs = 1,
```

Comment:
Is this because multiple chains/threads are handled directly by cmdstan instead?

Reply:
Yep!